### PR TITLE
Biogenerators can be unwrenched

### PR DESCRIPTION
**Pull request by carlarctg**
## About The Pull Request

Biogenerators can be unwrenched

## Why It's Good For The Game

If significantly more round impacting things like smoke machines and hydro trays that drop plants that turn into meth bath salts krokodil smoke can be unwrenched i dont see why this cant

## Changelog

:cl:
qol: Biogenerators can be unwrenched
/:cl:

### CODE REVIEW

**Review by san7890 — CHANGES_REQUESTED** [review]
> > I believe it is used as a wall machine on some maps, so you should probably copy the logic from smartfridge to make it require unwelding before unwrenching.
> 
> nah

you should at the very least copy the atmos blocking logic from the smartfridge because otherwise lemoninthedark will smash your skull in with a wrench. i don't recall if biogenerators are still placed within walls, but if they are you'll likely have to fix that across the maps because that is really stupid. you can ignore this if they aren't placed in walls, but i think they should still have atmos blocking some way (even if it's not in this PR)

**Comment by carlarctg** [comment]
> > > I believe it is used as a wall machine on some maps, so you should probably copy the logic from smartfridge to make it require unwelding before unwrenching.
> > 
> > 
> > nah
> 
> you should at the very least copy the atmos blocking logic from the smartfridge because otherwise lemoninthedark will smash your skull in with a wrench. i don't recall if biogenerators are still placed within walls, but if they are you'll likely have to fix that across the maps because that is really stupid. you can ignore this if they aren't placed in walls, but i think they should still have atmos blocking some way (even if it's not in this PR)

siiiiiiiiigh.............................................

**Comment by carlarctg** [comment]
@san7890

File: code/modules/hydroponics/biogenerator.dm [hunk 1]
```diff
@@ -44,13 +44,60 @@
...
+		to_chat(user, span_warning("[src] needs to be wrenched to the floor!"))
+		return TRUE
+	if(!tool.tool_start_check(user, amount=2))
+		return TRUE
+	user.visible_message(
+		span_notice("[user.name] starts to weld the [name] to the floor."),
+		span_notice("You start to weld [src] to the floor..."),
+		span_hear("You hear welding."),
+	)
+	if(!tool.use_tool(src, user, delay=100, volume=100))
```
Comment (Ghommie) [inline code/modules/hydroponics/biogenerator.dm:95]:
```suggestion
	if(!tool.use_tool(src, user, 10 SECONDS, volume=100))
```
10 seconds seem a tad much, but I'm not against the option of securing it in place in any way.

File: code/modules/hydroponics/biogenerator.dm [hunk 2]
```diff
@@ -44,13 +44,60 @@
...
+	..()
+	if(welded_down)
+		if(!tool.tool_start_check(user, amount=2))
+			return TRUE
+		user.visible_message(
+			span_notice("[user.name] starts to cut the [name] free from the floor."),
+			span_notice("You start to cut [src] free from the floor..."),
+			span_hear("You hear welding."),
+		)
+		if(!tool.use_tool(src, user, delay=100, volume=100))
```
Comment (Ghommie) [inline code/modules/hydroponics/biogenerator.dm:80]:
```suggestion
		if(!tool.use_tool(src, user, 10 SECONDS, volume=100))
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Pull request merged**